mysql-test/t/percona_log_slow_admin_statements_sp.test

```diff
@@ -0,0 +1,45 @@
+#
+# Test log_slow_slave_admin_statements
+#
+
+--disable_warnings
+DROP TABLE IF EXISTS t1;
+--enable_warnings
+
+CREATE TABLE t1 (a INT);
+
+SET SESSION min_examined_row_limit=0;
+SET SESSION long_query_time=0;
+
+delimiter ^;
+CREATE PROCEDURE test()
+BEGIN
+   INSERT INTO t1 VALUES(0);
+   CREATE INDEX i ON t1(a);
+   INSERT INTO t1 VALUES(1);
+END^
+delimiter ;^
+
+#
+# Test enabled admin statement slow-logging
+#
+SET GLOBAL log_slow_admin_statements=OFF;
+SET GLOBAL log_slow_sp_statements=ON;
+
+--let log_file=percona_log_slow_admin_stmt_sp_1
+--source include/log_start.inc
+call test();
+--source include/log_stop.inc
+--let grep_pattern=INSERT INTO t1 VALUES\(1\);
+--source include/log_grep.inc
+
+SET GLOBAL log_slow_admin_statements=default;
+SET SESSION long_query_time=default;
```

Comment:
- No need to restore session variables to their defaults;
- log_slow_sp_statements is not restored, should be;
- global variables should be restored not by assigning the default value, but by saving the original value first and restoring to that (so that mtr --mysqld=--global-var=non-default works)